src/workerd/api/web-socket.c++

```diff
@@ -185,8 +185,10 @@ jsg::Ref<WebSocket> WebSocket::constructor(jsg::Lock& js,
 
   // Check if we have a valid URL
   constexpr auto wsErr = "WebSocket Constructor: "_kj;
-  kj::Url urlRecord =
-      JSG_REQUIRE_NONNULL(kj::Url::tryParse(url), DOMSyntaxError, wsErr, "The url is invalid.");
+  // Any authority state (username and/or password) members should be removed.
+  // Users should use Authorization header for this purpose.
+  kj::Url urlRecord = JSG_REQUIRE_NONNULL(kj::Url::tryParse(url, kj::Url::HTTP_PROXY_REQUEST),
```

Comment:
This isn't correct. Please make this constructor do what fetch() does.

Reply:
Specifically it first parses as REMOTE_HREF and then re-stringifies the URL as HTTP_PROXY_REQUEST, with a bunch of other logic in between. Please actually look carefully at it and make sure we're doing the same logic here.